spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/repository/support/SimpleDatastoreRepository.java

```diff
@@ -218,7 +230,95 @@ public void deleteAllById(Iterable<? extends I> iterable) {
 	}
 
 	@Override
-	public <S extends T, R> R findBy(Example<S> example, Function<FluentQuery.FetchableFluentQuery<S>, R> queryFunction) {
-		throw new UnsupportedOperationException();
+	public <S extends T, R> R findBy(Example<S> example,
+			Function<FluentQuery.FetchableFluentQuery<S>, R> queryFunction) {
+		Assert.notNull(example, "Example must not be null!");
+		Assert.notNull(queryFunction, "Query function must not be null!");
+
+		return queryFunction.apply(new DatastoreFluentQueryByExample<>(example, example.getProbeType()));
+	}
+
+	class DatastoreFluentQueryByExample<S extends T, R> implements FluentQuery.FetchableFluentQuery<R> {
+		private final Example<S> example;
+
+		private final Sort sort;
+
+		private final Class<?> domainType;
+
+		private final Class<R> resultType;
+
+		DatastoreFluentQueryByExample(Example<S> example, Class<R> resultType) {
+			this(example, Sort.unsorted(), resultType, resultType);
+		}
+
+		DatastoreFluentQueryByExample(Example<S> example, Sort sort, Class<?> domainType, Class<R> resultType) {
+			this.example = example;
+			this.sort = sort;
+			this.domainType = domainType;
+			this.resultType = resultType;
+		}
+
+		@NonNull
+		@Override
+		public FetchableFluentQuery<R> sortBy(@NonNull Sort sort) {
+			return new DatastoreFluentQueryByExample<>(this.example, sort, this.domainType, this.resultType);
+		}
+
+		@NonNull
+		@Override
+		public Optional<R> one() {
+			return (Optional<R>) SimpleDatastoreRepository.this.findOne(this.example);
+		}
+
+		@Nullable
+		@Override
+		public R oneValue() {
+			Optional<R> one = one();
+			return one.orElse(null);
+		}
+
+		@Override
+		public R firstValue() {
+			return (R) SimpleDatastoreRepository.this.findFirstSorted(this.example, this.sort);
```

Comment:
Should this assert that the current sort is not `Sort.unsorted()` before trying to find first sorted? Or at least log a warning. So users don't have to troubleshoot non-repeatable behavior...